File: backend/src/main/java/pt/ulisboa/tecnico/rnl/dei/dms/evaluation/dto/StudentProfileDto.java
package pt.ulisboa.tecnico.rnl.dei.dms.evaluation.dto;

import java.time.LocalDate;
import java.util.List;

public record StudentProfileDto(long personId, String personName, List<EnrolledUcDto> ucs, List<PendingEvaluationDto> pendingEvaluations) {
    public record EnrolledUcDto(long ucId, String ucName, Double average) {}

    public record PendingEvaluationDto(String type, long id, String title, LocalDate deadline, String ucName) {}
}

File: frontend/src/models/evaluation/StudentProfileDto.ts
export interface EnrolledUcDto {
  ucId: number
  ucName: string
  average: number | null
}

export interface PendingEvaluationDto {
  type: string
  id: number
  title: string
  deadline: string
  ucName: string
}

export default class StudentProfileDto {
  personId?: number
  personName?: string
  ucs?: EnrolledUcDto[]
  pendingEvaluations?: PendingEvaluationDto[]

  constructor(obj?: Partial<StudentProfileDto>) {
    Object.assign(this, obj)
  }
}